lib/functions/general/git.sh

```diff
@@ -55,8 +55,10 @@ function improved_git_fetch() {
 function git_ensure_safe_directory() {
 	if [[ -n "$(command -v git)" ]]; then
 		local git_dir="$1"
-		display_alert "git: Marking all directories as safe, which should include" "$git_dir" "debug"
-		regular_git -C "$1" config --local --get safe.directory "$1" > /dev/null || regular_git -C "$1" config --local --add safe.directory "$1"
+		if regular_git -C "$1" -C rev-parse --git-dir > /dev/null 2>&1; then
+			display_alert "git: Marking all directories as safe, which should include" "$git_dir" "debug"
+			regular_git -C "$1" config --local --get safe.directory "$1" > /dev/null || regular_git -C "$1" config --local --add safe.directory "$1"
+		fi
```

Comment:
@igorpecovnik Same thought as my prev comment https://github.com/armbian/build/pull/6206#discussion_r1468158352 regarding `"$1"` to be replaced by `"$git_dir"` all the way for consistency.